packages/smooth_app/lib/pages/preferences/country_selector/openfoodfacts_country_name_extension.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:l10n_countries/l10n_countries.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/pages/preferences/country_selector/tmp_country_iso3.dart';
+
+extension OpenFoodFactsCountryNameExtension on OpenFoodFactsCountry {
+  String? getLocalizedName(OpenFoodFactsLanguage locale) {
+    final String? iso3 = tmpCountryIso3[this];
+    if (iso3 == null) {
+      return null;
+    }
+
+    final CountriesLocaleMapper mapper = CountriesLocaleMapper();
+    final LocaleMap result = mapper.localize(
+      {iso3},
+      mainLocale: locale,
+    );
+
+    return result.values.firstOrNull;
+  }
+
+  String getEnglishName() =>
+      getLocalizedName(OpenFoodFactsLanguage.ENGLISH) ??
+      toString().split('.').last;
```

Comment:
We're never using `getEnglishName` in the rest of the code, are we?
Then we should put the fallback naming in the regular `getLocalizedName` method, using the previous code, something like:
```dart
        String? name = result.values.firstOrNull;
        if (language != OpenFoodFactsLanguage.ENGLISH || name != null) {
          return name;
        }
        name = this
            .toString()
            .replaceAll('OpenFoodFactsCountry.', '')
            .replaceAll('_', ' ');
        return
            '${name[0].toUpperCase()}${name.substring(1).toLowerCase()}';
```